diff --git a/CMakeLists.txt b/CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -692,14 +692,14 @@
 endif(UNIX)
 
 if(USE_PYTHON AND NOT SKBUILD)
   # install egg file to let python/pip know that Netgen ist installed
-  file( WRITE "netgen_mesher-py3.egg-info"
+  file( WRITE "netgen_mesher-${NETGEN_VERSION_MAJOR}.${NETGEN_VERSION_MINOR}.${NETGEN_VERSION_PATCH}.post${NETGEN_VERSION_TWEAK}.egg-info"
 "Metadata-Version: 2.1
 Name: netgen-mesher
 Version: ${NETGEN_VERSION_MAJOR}.${NETGEN_VERSION_MINOR}.${NETGEN_VERSION_PATCH}.post${NETGEN_VERSION_TWEAK}
 ")
-  install(FILES netgen_mesher-py3.egg-info DESTINATION ${NG_INSTALL_DIR_PYTHON} COMPONENT netgen)
+  install(FILES netgen_mesher-${NETGEN_VERSION_MAJOR}.${NETGEN_VERSION_MINOR}.${NETGEN_VERSION_PATCH}.post${NETGEN_VERSION_TWEAK}.egg-info DESTINATION ${NG_INSTALL_DIR_PYTHON} COMPONENT netgen)
 endif()
 
 if(APPLE AND NOT SKBUILD)
     # create some auxiliary files